contentful/apps
# feat: add hosted app actions for sap [EXT-5612]

## Purpose

We will use hosted app actions to make api calls with the AIR header for the SAP AIR app. This PR introduces a lot of the boilerplate around hosted app actions.

## Approach

Updated the SAP app to follow the project structure and boilerplate used in the MS Teams and Vercel apps. See this PR as an example for Vercel: https://github.com/contentful/apps/pull/6769

Specifically, here is a summary of the changes:

- Move all frontend app code into a new `frontend` directory
    - Update the `vite.config` to support parent project build outDir location
- Introduce a new `app-actions` directory
    - Copied the 3 hosted app actions used in this old branch: https://github.com/contentful/apps/tree/feat/sap-air-original
    - Added testing and other boilerplate used in other apps for HAAs
- Set up top level project to handle the double deploy and build for both apps (public marketplace and private AIR header)
    - Add manifest file for the new actions
    - Add build actions and call actions scripts

## Testing steps

I confirmed that I could successfully build the frontend and actions and deployed to both staging apps. Confirmed that the app config page and field locations are rendering for both staging apps. I am not able to see products in the Dialog, but that's also a problem I am seeing on our feature branch.

<img width="922" alt="Screenshot 2024-08-16 at 12 38 10 PM" src="https://github.com/user-attachments/assets/c2e070b0-cd46-42c4-ad1f-7cca1d40d5df">


## Breaking Changes



## Dependencies and/or References



## Deployment
